spikeinterface/sortingcomponents/motion_estimation.py

```diff
@@ -180,91 +181,176 @@ class DecentralizedRegistration:
                                                   torch_device=torch_device, batch_size=batch_size,
                                                   max_displacement_um=max_displacement_um,
                                                   corr_threshold=corr_threshold, time_horizon_s=time_horizon_s,
                                                   bin_duration_s=bin_duration_s, progress_bar=False)
-            if extra_check:
+
+            if spatial_prior:
+                all_pairwise_displacements[i] = pairwise_displacement
+                all_pairwise_displacement_weights[i] = pairwise_displacement_weight
+
+            if extra_check is not None:
                 extra_check['pairwise_displacement_list'].append(pairwise_displacement)
 
             if verbose:
                 print(f'Computing global displacement: {i + 1} / {len(non_rigid_windows)}')
 
-            motion[:, i] = compute_global_displacement(pairwise_displacement,
-                                                       pairwise_displacement_weight=pairwise_displacement_weight,
-                                                       convergence_method=convergence_method,
-                                                       robust_regression_sigma=robust_regression_sigma,
-                                                       lsqr_robust_n_iter=lsqr_robust_n_iter, progress_bar=False)
+            # TODO: if spatial_prior, do this after the loop
+            if not spatial_prior:
+                motion[:, i] = compute_global_displacement(
+                    pairwise_displacement,
+                    pairwise_displacement_weight=pairwise_displacement_weight,
+                    convergence_method=convergence_method,
+                    robust_regression_sigma=robust_regression_sigma,
+                    lsqr_robust_n_iter=lsqr_robust_n_iter,
+                    temporal_prior=temporal_prior,
+                    spatial_prior=spatial_prior,
+                    soft_weights=soft_weights,
+                    progress_bar=False,
+                )
+
+        if spatial_prior:
+            motion = compute_global_displacement(
+                all_pairwise_displacements,
+                pairwise_displacement_weight=all_pairwise_displacement_weights,
+                convergence_method=convergence_method,
+                robust_regression_sigma=robust_regression_sigma,
+                lsqr_robust_n_iter=lsqr_robust_n_iter,
+                temporal_prior=temporal_prior,
+                spatial_prior=spatial_prior,
+                soft_weights=soft_weights,
+                progress_bar=False,
+            )
+        elif len(non_rigid_windows) > 1:
+            # if spatial_prior is False, we still want keep the spatial bins
+            # correctly offset from each other
+            for i in range(len(non_rigid_windows) - 1):
+                motion[:, i + 1] -= np.median(motion[:, i + 1] - motion[:, i])
```

Comment:
Should we also make sure that np.median(motion[:, 0]) == 0 ?
On short the first layer do not move in median. and then the diff is 0 relative to the previous ?

Reply:
Forget it this is done just afte rthis.